### PR TITLE
Pdo driver interface/password sensitive param

### PR DESCRIPTION
**Pull request by heap-s**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | #872

### CODE REVIEW

File: src/Driver/Pdo/PdoDriverInterface.php [hunk 1]
```diff
@@ -78,7 +78,7 @@ public function getUsername(): string;
      *
      * @param string $password The password for establishing DB connection.
      */
-    public function password(string $password): void;
+    public function password(#[\SensitiveParameter] string $password): void;
```
Comment (vjik) [inline src/Driver/Pdo/PdoDriverInterface.php:81]: Attribute not need in interface. Attribute should be added in implementation. Example: https://3v4l.org/KNALl

**Review by vjik — CHANGES_REQUESTED** [review]

File: src/Driver/Pdo/PdoDriverInterface.php (hunk 1 above)
Comment (heap-s) [reply src/Driver/Pdo/PdoDriverInterface.php:81]: So you'd prefer to see this in AbstractPdoDriver class that implements PdoDriverInterface?

Comment (vjik) [reply src/Driver/Pdo/PdoDriverInterface.php]: Yes

Comment (heap-s) [reply src/Driver/Pdo/PdoDriverInterface.php]: I believe this is resolved, thanks for the help!

Comment (vjik) [reply src/Driver/Pdo/PdoDriverInterface.php]: Add to constructor parameter also. And need line in changelog.

File: CHANGELOG.md [hunk 2]
```diff
@@ -34,6 +34,7 @@
 - Enh #862: Refactor PHP type of `ColumnSchemaInterface` instances (@Tigrov)
 - Enh #865: Raise minimum PHP version to `^8.1` with minor refactoring (@Tigrov, @vjik)
 - Enh #798: Allow `QueryInterface::one()` and `QueryInterface::all()` to return objects (@darkdef, @Tigrov)
+- Chg #872: Add `#[\SensitiveParameter]` attribute for password (@heap-s)
```
Comment (vjik) [inline CHANGELOG.md:37]:
```suggestion
- Enh #872: Use `#[\SensitiveParameter]` attribute to mark sensitive parameters (@heap-s)
```

**Review by vjik — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (heap-s) [reply CHANGELOG.md]: I believe that I've made all of your suggested changes, sorry this has been quite the process. Now that I understand the process for contributing a bit better here, I can make this easier in the future.

**Review by samdark — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (vjik) [reply CHANGELOG.md]: It's OK. Thanks for contribution 👍

**Comment by Tigrov** [comment]
Can we test it?

Does it make sence to add it to the interface?

https://github.com/yiisoft/db/blob/99d69c17679dd306059c19797bd587b70582a875/src/Driver/Pdo/PdoDriverInterface.php#L81

**Comment by samdark** [comment]
@Tigrov not really.

**Comment by vjik** [comment]
Hm, if we can throw exception, then we can check trace. Example: https://3v4l.org/OYZAt

**Comment by Tigrov** [comment]
To throw an exception we can pass an integer value instead of string

**Comment by heap-s** [comment]
Have we decided what we're doing here in terms of testing?

**Comment by Tigrov** [comment]
> Have we decided what we're doing here in terms of testing?

Can you try to use this example for testing the SensitiveParameter?

https://3v4l.org/Gh1lY

**Comment by heap-s** [comment]
@Tigrov The solution that I think of implementing is https://3v4l.org/QK6Yo I believe that this is appropriate and what you'd like in terms of testing for the SensitiveParameter. Let me know what you think.

**Comment by Tigrov** [comment]
> Let me know if you'd like me to squash these commits, so that the history is cleaner

Let's see on test results then squash and merge it.

**Review by vjik — APPROVED** [review]

File: tests/Db/Driver/PDO/PDODriverTest.php [hunk 3]
```diff
@@ -62,4 +62,23 @@ public function testGetUsername(): void
...
+            new PDODriver($dsn, password: null);
+        } catch (\TypeError $e) {
+            $this->assertTrue($e->getTrace()[0]['args'][2] instanceof \SensitiveParameterValue);
+        }
+        $pdoDriver = new PDODriver($dsn);
+        try {
+            $pdoDriver->password(null);
+        } catch (\TypeError $e) {
+            $this->assertTrue($e->getTrace()[0]['args'][0] instanceof \SensitiveParameterValue);
+        }
```
Comment (Tigrov) [inline tests/Db/Driver/PDO/PDODriverTest.php:82]:
```suggestion
        }
```

**Review by Tigrov — APPROVED** [review]

**Comment by Tigrov** [comment]
Thanks 👍

**Pull request merged**